data_types/benches/benchmark.rs

```diff
@@ -0,0 +1,624 @@
+                                                    }
+                                                    let tag_index =
+                                                        *partition.dict.get(tag).unwrap();
+
+                                                    Value::Tag(tag_index)
+                                                }
+                                                wb::ColumnValue::F64Value => {
+                                                    let val =
+                                                        value.value_as_f64value().unwrap().value();
+
+                                                    Value::F64(val)
+                                                }
+                                                wb::ColumnValue::I64Value => {
+                                                    let val =
+                                                        value.value_as_i64value().unwrap().value();
+
+                                                    Value::I64(val)
+                                                }
+                                                _ => panic!("not supported!"),
+                                            };
+
+                                            let column_value = ColumnValue {
+                                                column_name_index: column_index,
+                                                value: val,
+                                            };
+
+                                            row.values.push(column_value);
+                                        }
+
+                                        table.rows.push(row);
+                                    }
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    fn partition_count(&self) -> usize {
+        self.partitions.len()
+    }
+
+    fn row_count(&self) -> usize {
+        let mut count = 0;
+
+        for p in self.partitions.values() {
+            for t in p.tables.values() {
+                count += t.rows.len();
+            }
+        }
+
+        count
+    }
+
+    fn tag_cardinality(&self) -> usize {
+        let mut tags = BTreeMap::new();
+
+        for p in self.partitions.values() {
+            for t in p.tables.values() {
+                for r in &t.rows {
+                    for v in &r.values {
+                        if let Value::Tag(idx) = &v.value {
+                            tags.insert(*idx, ());
+                        }
+                    }
+                }
+            }
+        }
+
+        tags.len()
+    }
+
+    fn measurement_count(&self) -> usize {
+        let mut measurements = BTreeMap::new();
+
+        for p in self.partitions.values() {
+            for t in p.tables.values() {
+                measurements.insert(t.name.to_string(), ());
+            }
+        }
```

Comment:
This is pretty much what `BTreeSet` does; I'd recommend using that:

```suggestion
        let mut measurements = BTreeSet::new();

        for p in self.partitions.values() {
            for t in p.tables.values() {
                measurements.insert(t.name.to_string());
            }
        }
```

Reply:
Added in 5769558cc38be26d503975020203f059b7376ef8 in #607 as this PR will be getting closed out.